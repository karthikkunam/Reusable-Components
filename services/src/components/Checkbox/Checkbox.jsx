import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

const Checkbox = props => {
  const [checked, setChecked] = useState(props.checked);
  const { id, disabled, className, label, onChange, value, children, checked: checkedProp } = props;

  useEffect(() => {
    setChecked(checkedProp);
  }, [checkedProp]);

  const handleCheckboxClick = e => {
    const {
      target: { checked },
    } = e;
    if (disabled) {
      return;
    }
    onChange && onChange(checked, value, e);
    setChecked(checked);
  };

  return (
    <fieldset className={`checkbox mt-1 ${className}`}>
      <label id={id} className={`checkbox-label ${disabled ? 'disabled' : ''}`}>
        <span className="checkbox-text">{label ? label : children}</span>
        <input type="checkbox" disabled={disabled} checked={checked ? 'checked' : ''} onChange={handleCheckboxClick} />
        <span className="checkmark"></span>
      </label>
    </fieldset>
  );
};

Checkbox.propTypes = {
  id: PropTypes.string.isRequired,
  className: PropTypes.string,
  label: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  checked: PropTypes.bool,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.any,
  children: PropTypes.node,
};

Checkbox.defaultProps = {
  className: '',
  children: '',
};

export default Checkbox;
