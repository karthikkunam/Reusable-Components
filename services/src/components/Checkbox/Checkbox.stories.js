import React from 'react';
import { action } from '@storybook/addon-actions';
import { withKnobs } from '@storybook/addon-knobs';
import { boolean } from '@storybook/addon-knobs';
import Checkbox from '.';

export default {
  title: 'Checkbox',
  component: Checkbox,
  decorators: [withKnobs]
};

export const CheckboxButton = () => (
  <>
    <div> CheckBoxes</div>
    <Checkbox id="dropdown1" label="one" value="one" onChange={action('checked')} />
    <Checkbox id="dropdown2" label="Two" value="two" onChange={action('checked')} />
    <Checkbox id="dropdown3" label="Three" value="three" onChange={action('checked')} />
    <Checkbox id="dropdown4" label="Four" value="four" onChange={action('checked')} />
  </>
);
export const CheckedCheckbox = () => (
  <>
    <div> CheckBoxes with default checked</div>
    <Checkbox id="dropdown1" label="one" checked={true} value="one" onChange={action('checked')} />
    <Checkbox id="dropdown2" label="Two" checked={true} value="two" onChange={action('checked')} />
    <Checkbox id="dropdown2" label="Three" checked={false} value="three" onChange={action('checked')} />
  </>
);
export const DisabledCheckbox = () => (
  <>
    <div> Disabled CheckBox</div>
    <Checkbox id="dropdown1" label="one" checked={false} value="label" onChange={action('checked')} />
    <Checkbox id="dropdown2" label="Two" disabled={true} value="label" onChange={action('checked')} />
  </>
);

export const CheckboxWithoutLabel = () => (
  <>
    <div> CheckBox label from children</div>
    <Checkbox id="dropdown1" checked={false} value="label" onChange={action('checked')}>
      Children
    </Checkbox>
    <Checkbox id="dropdown2" checked={true} value="label" onChange={action('checked')}>
      Children
    </Checkbox>
  </>
);

export const PlayCheckbox = () => (
  <>
    <Checkbox id="dropdown1" label="one" checked={boolean('Checked', false, 'GROUP-ID1')} value="one" onChange={action('checked')} />
  </>
);