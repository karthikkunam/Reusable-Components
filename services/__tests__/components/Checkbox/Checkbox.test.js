import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from '../../../src/components/Checkbox';

describe('Checkbox Component', () => {
  it('renders correctly', () => {
    const props = {
      id: '',
      disabled: false,
      className: '',
      label: '',
    };
    const component = shallow(<Checkbox {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders handleCheckboxClick correctly', () => {
    const props = {
      id: 'dropdown1',
      disabled: false,
      className: '',
      label: 'One',
      onChange: jest.fn(),
    };
    const component = shallow(<Checkbox {...props} />);
    component
      .find('label')
      .at(0)
      .simulate('change', { target: { checked: true } });
    expect(component).toMatchSnapshot();
  });
  it('should return null when disabled is true onclick of handleCheckboxClick', () => {
    const props = {
      id: 'dropdown1',
      disabled: true,
      className: '',
      label: 'One',
      onChange: jest.fn(),
    };
    const component = shallow(<Checkbox {...props} />);
    component
      .find('label')
      .at(0)
      .simulate('change', { target: { checked: false } });
    expect(component).toMatchSnapshot();
  });
  it('renders Checkbox with default checked', () => {
    const props = {
      id: 'dropdown1',
      disabled: false,
      className: '',
      label: 'One',
      onChange: jest.fn(),
      checked: true,
    };
    const component = shallow(<Checkbox {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders Checkbox with checked and disable', () => {
    const props = {
      id: 'dropdown1',
      disabled: true,
      className: '',
      label: 'One',
      onChange: jest.fn(),
      checked: true,
    };
    const component = shallow(<Checkbox {...props} />);
    expect(component).toMatchSnapshot();
  });
});
