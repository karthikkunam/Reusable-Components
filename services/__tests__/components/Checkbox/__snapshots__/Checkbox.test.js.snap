// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component renders Checkbox with checked and disable 1`] = `
<fieldset
  className="checkbox mt-1 "
>
  <label
    className="checkbox-label disabled"
    id="dropdown1"
    onChange={[Function]}
  >
    <span
      className="checkbox-text"
    >
      One
    </span>
    <input
      checked="checked"
      type="checkbox"
    />
    <span
      className="checkmark"
    />
  </label>
</fieldset>
`;

exports[`Checkbox Component renders Checkbox with default checked 1`] = `
<fieldset
  className="checkbox mt-1 "
>
  <label
    className="checkbox-label "
    id="dropdown1"
    onChange={[Function]}
  >
    <span
      className="checkbox-text"
    >
      One
    </span>
    <input
      checked="checked"
      type="checkbox"
    />
    <span
      className="checkmark"
    />
  </label>
</fieldset>
`;

exports[`Checkbox Component renders correctly 1`] = `
<fieldset
  className="checkbox mt-1 "
>
  <label
    className="checkbox-label "
    id=""
    onChange={[Function]}
  >
    <span
      className="checkbox-text"
    />
    <input
      checked=""
      type="checkbox"
    />
    <span
      className="checkmark"
    />
  </label>
</fieldset>
`;

exports[`Checkbox Component renders handleCheckboxClick correctly 1`] = `
<fieldset
  className="checkbox mt-1 "
>
  <label
    className="checkbox-label "
    id="dropdown1"
    onChange={[Function]}
  >
    <span
      className="checkbox-text"
    >
      One
    </span>
    <input
      checked="checked"
      type="checkbox"
    />
    <span
      className="checkmark"
    />
  </label>
</fieldset>
`;

exports[`Checkbox Component should return null when disabled is true onclick of handleCheckboxClick 1`] = `
<fieldset
  className="checkbox mt-1 "
>
  <label
    className="checkbox-label disabled"
    id="dropdown1"
    onChange={[Function]}
  >
    <span
      className="checkbox-text"
    >
      One
    </span>
    <input
      checked=""
      type="checkbox"
    />
    <span
      className="checkmark"
    />
  </label>
</fieldset>
`;
